### PR TITLE
Windows variant of Linux installer without MSys2

### PR DESCRIPTION
**Pull request by otabuzzman**
This PR complements the PR for a Windows install script for the main TornadoVM repo. Details can be found there.

### CODE REVIEW

**Comment by jjfumero** [comment]
Thanks, I am getting an error during the execution of Level Zero on Windows 11. Did you do any extra step to enable Level Zero on the Intel Integrated Graphics?

**Comment by otabuzzman** [comment]
There were in fact three leve-zero DLLs already in my system32 folder:
ze_validation_layer.dll, ze_tracing_layer.dll, and ze_loader.dll. After
moving these away, zello_world.exe worked. I had no need to move any DLL
from the level zero repo.

Juan Fumero ***@***.***> schrieb am Mi. 20. März 2024 um
11:19:

**Comment by jjfumero** [comment]
Thank you, that did the trick. When I first installed it (a couple of years ago) this was not the case. It might be worth filing an issue in the Level Zero repo.

**Comment by otabuzzman** [comment]
Good idea. I stepped thru the API when I couldn’t get zello_world to search
PATH for its DLLs.  It calls
LoadLibraryExA(NAME, nullptr, LOAD_LIBRARY_SEARCH_SYSTEM32) and pointed me
at the DLLs that were already there. Forcing the DLLs to live in system32
might be ok in terms of security, but indeed worth a note in the repo. I
filed an issue:
https://github.com/oneapi-src/level-zero/issues/138

Juan Fumero ***@***.***> schrieb am Mi. 20. März 2024 um
18:42:

**Comment by jjfumero** [comment]
I can build the library on Windows using the MS Tools. However, I can an error when I run. I am pretty sure it is my fault and I am missing something:

```bash
cmake --build . --config Release
MSBuild version 17.8.3+195e7f5a3 for .NET Framework

  tornado-levelzero.vcxproj -> C:\Users\jjfum\source\repos\levelzero-jni\levelZeroLib\build\Release\tornado-levelzero.dll

C:\Users\jjfum\source\repos\levelzero-jni\levelZeroLib\build>cd ../..

C:\Users\jjfum\source\repos\levelzero-jni>.\scripts\run.cmd

C:\Users\jjfum\source\repos\levelzero-jni>java -Djava.library.path=./levelZeroLib/build/Release -cp target/beehive-levelzero-jni-0.1.2.jar uk.ac.manchester.tornado.drivers.spirv.levelzero.samples.TestLevelZero copyData.spv
Testing Level-ZERO JNI Library
Exception in thread "main" java.lang.UnsatisfiedLinkError: C:\Users\jjfum\source\repos\levelzero-jni\levelZeroLib\build\Release\tornado-levelzero.dll: Can't find dependent libraries
        at java.base/jdk.internal.loader.NativeLibraries.load(Native Method)
        at java.base/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:331)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:197)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:139)
        at java.base/jdk.internal.loader.NativeLibraries.findFromPaths(NativeLibraries.java:259)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:251)
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2451)
        at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:916)
        at java.base/java.lang.System.loadLibrary(System.java:2059)
        at uk.ac.manchester.tornado.drivers.spirv.levelzero.LevelZeroDriver.<clinit>(LevelZeroDriver.java:51)
        at uk.ac.manchester.tornado.drivers.spirv.levelzero.samples.TestLevelZero.main(TestLevelZero.java:105)
```

Any ideas? I used to get these errors when I built the library and I have a missing name when invoking a native function, but now it seems the error is when loading the DLL.

File: levelZeroLib/CMakeLists.txt
Comment (jjfumero) [inline levelZeroLib/CMakeLists.txt:42]: Does this mean we need `sccache` installed as a dependency?

**Comment by jjfumero** [comment]
Can you please sync with the latest `develop` branch? So the diff just shows the new changes.

File: levelZeroLib/src/levelZeroContext.cpp
Comment (jjfumero) [inline levelZeroLib/src/levelZeroContext.cpp:112]: I think we should also use `int64_t` in Linux as well. Let's keep it for now, I will make a note and do this after we merge, so we can unify the Windows and Linux ports as much as possible.

**Comment by otabuzzman** [comment]
Try adding the folder containing the Intel Level Zero DLLs to PATH (e.g.
set PATH=<somedir>\level-zero\build\bin\Release;%PATH%). I put this into
the installer script but obviously forgot to mention it in the README.md.

Juan Fumero ***@***.***> schrieb am Do. 21. März 2024 um
06:57:

**Comment by otabuzzman** [comment]
I don‘t know if we need it but I guess we do not. To be honest, I simply
copied (with rather few modifications) the lines to compile for Windows
from Intel's CMakeLists.txt in the level zero repo.

Juan Fumero ***@***.***> schrieb am Do. 21. März 2024 um
08:15:

**Comment by jjfumero** [comment]
Same behaviour on my system:

```bash
C:\Users\jjfum\source\repos\levelzero-jni>set ZE_SHARED_LOADER=C:\Users\jjfum\source\repos\level-zero\build\lib\release\ze_loader.lib

C:\Users\jjfum\source\repos\levelzero-jni>set CPLUS_INCLUDE_PATH=C:\Users\jjfum\source\repos\level-zero\include

C:\Users\jjfum\source\repos\levelzero-jni>set C_INCLUDE_PATH=C:\Users\jjfum\source\repos\level-zero\include

C:\Users\jjfum\source\repos\levelzero-jni>set PATH=C:\Users\jjfum\source\repos\levelzero-jni\levelZeroLib\build\Release;%PATH%

C:\Users\jjfum\source\repos\levelzero-jni> dir C:\Users\jjfum\source\repos\levelzero-jni\levelZeroLib\build\Release

 Directory of C:\Users\jjfum\source\repos\levelzero-jni\levelZeroLib\build\Release

21/03/2024  08:55    <DIR>          .
21/03/2024  08:55    <DIR>          ..
21/03/2024  08:55           303,104 tornado-levelzero.dll
21/03/2024  08:55            36,546 tornado-levelzero.exp
21/03/2024  08:55            62,140 tornado-levelzero.lib
21/03/2024  08:55         6,533,120 tornado-levelzero.pdb
               4 File(s)      6,934,910 bytes
               2 Dir(s)  434,048,061,440 bytes free

C:\Users\jjfum\source\repos\levelzero-jni> java -Djava.library.path=./levelZeroLib/build/Release -cp target/beehive-levelzero-jni-0.1.2.jar uk.ac.manchester.tornado.drivers.spirv.levelzero.samples.TestLevelZero copyData.spv
Testing Level-ZERO JNI Library
Exception in thread "main" java.lang.UnsatisfiedLinkError: C:\Users\jjfum\source\repos\levelzero-jni\levelZeroLib\build\Release\tornado-levelzero.dll: Can't find dependent libraries
        at java.base/jdk.internal.loader.NativeLibraries.load(Native Method)
        at java.base/jdk.internal.loader.NativeLibraries$NativeLibraryImpl.open(NativeLibraries.java:331)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:197)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:139)
        at java.base/jdk.internal.loader.NativeLibraries.findFromPaths(NativeLibraries.java:259)
        at java.base/jdk.internal.loader.NativeLibraries.loadLibrary(NativeLibraries.java:251)
        at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2451)
        at java.base/java.lang.Runtime.loadLibrary0(Runtime.java:916)
        at java.base/java.lang.System.loadLibrary(System.java:2059)
        at uk.ac.manchester.tornado.drivers.spirv.levelzero.LevelZeroDriver.<clinit>(LevelZeroDriver.java:51)
        at uk.ac.manchester.tornado.drivers.spirv.levelzero.samples.TestLevelZero.main(TestLevelZero.java:105)
```

**Comment by otabuzzman** [comment]
I think your PATH still misses the folder with the Intel DLLs. Derived from
your value for ZE_SHARED_LOADER the appropriate folder for your setup might
be C:\Users\jjfum\source\repos\level-zero\build\bin\release (mind bin
instead of lib). This PATH setup might work:


set PATH=
C:\Users\jjfum\source\repos\level-zero\build\bin\release;C:\Users\jjfum\source\repos\levelzero-jni\levelZeroLib\build\Release;%PATH%


If that still does not work, look for a folder in
C:\Users\jjfum\source\repos\level-zero that contains DLLs whose names start
with ze_ and add it to PATH


Juan Fumero ***@***.***> schrieb am Do. 21. März 2024 um
10:42:

**Comment by jjfumero** [comment]
Thank you @otabuzzman , That made the trick. 

```bash
C:\Users\jjfum\source\repos\levelzero-jni>set PATH=C:\Users\jjfum\source\repos\level-zero\build\bin\Release;%PATH%

C:\Users\jjfum\source\repos\levelzero-jni>.\scripts\run.cmd

C:\Users\jjfum\source\repos\levelzero-jni>java -Djava.library.path=./levelZeroLib/build/Release -cp target/beehive-levelzero-jni-0.1.2.jar uk.ac.manchester.tornado.drivers.spirv.levelzero.samples.TestLevelZero copyData.spv
Testing Level-ZERO JNI Library
Driver Version: 17001183
Level Zero API Version:
=========================
Device Properties
=========================
STye                : ZE_STRUCTURE_TYPE_DEVICE_PROPERTIES
pNext               : 0
Type                : ZE_DEVICE_TYPE_GPU
vendorId            : 32902
deviceId            : 18048
flags               : 1
subdeviceId         : 0
coreClockRate       : 1500
maxMemAllocSize     : 4294959104
maxHardwareContext  : 65536
maxCommandQueuePriority: 0
numThreadsPerEU     : 7
physicalEUSimdWidth : 8
numEUsPerSubslice   : 8
numSubslicesPerSlice: 4
numSlices           : 1
timerResolution     : 52
timestampValidBits  : 36
kernelTimestampValidBits: 32
uuid                : [134, 128, 128, 70, 12, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0]
name                : Intel(R) UHD Graphics 770
```

Can you please complete the README file to add these instructions?

**Comment by jjfumero** [comment]
BTW, I used the latest version of Level Zero, so no need to checkout the tag `1.4`

**Comment by otabuzzman** [comment]
Great. I will update README.md tonight with access to my notebook
(currently traveling with smartphone only).

Juan Fumero ***@***.***> schrieb am Do. 21. März 2024 um
11:23:

**Review by jjfumero — APPROVED** [review]
Thank you @otabuzzman . This PR looks good to me.

**Comment by jjfumero** [comment]
Let's wait until @stratika can also reproduce it and then we merge.

**Review by stratika — APPROVED** [review]
LGTM, only a small typo in CHANGELOG.md.

**Comment by stratika** [comment]
@jjfumero, once we do the release of 0.1.3, we need to update also the path for the target directory that will have the new jar files.

**Pull request merged**
